File: Difficulty/Easy/[121] Best Time to Buy and Sell Stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2: return 0
        maxProfit, min_val = 0, prices[0]
        for i in range(1, len(prices)):
            if prices[i] < min_val:
                min_val = prices[i]
            else:
                maxProfit = max(maxProfit, prices[i] - min_val)
        return maxProfit

File: Difficulty/Easy/[121] Best Time to Buy and Sell Stock/solution.cpp
/*
 * @lc app=leetcode id=121 lang=cpp
 *
 * [121] Best Time to Buy and Sell Stock
 */

// @lc code=start
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() < 2) return 0;
        int maxProfit = 0, min_val = prices[0];
        for (int i = 1; i < prices.size(); ++i){
            if (prices[i] < min_val) {
                min_val = prices[i];
            }else{
                maxProfit = max(maxProfit, prices[i] -  min_val);
            }
        }
        return maxProfit;
    }
};
// @lc code=end

